source/extensions/omni.isaac.orbit/omni/isaac/orbit/sim/utils.py

```diff
@@ -777,3 +777,47 @@ def find_global_fixed_joint_prim(
                 break
 
     return fixed_joint_prim
+
+
+"""
+USD Variants.
+"""
+
+
+def set_usd_variants(prim_path: str, variants: object | dict[str, str], stage: Usd.Stage | None = None) -> None:
+    """
+    Sets the variant selections for the specified variant sets on a USD prim.
+
+    Args:
+        prim_path (str): The path of the USD prim.
+        variants (dict[str, str]): A dictionary mapping variant set names to variant selections.
+        stage (Usd.Stage | None, optional): The USD stage. If not provided, the current stage will be used.
+
+    Returns:
+        None
+    """
+    if stage is None:
+        stage = stage_utils.get_current_stage()
+
+    prim = stage.GetPrimAtPath(prim_path)
+
+    existing_variant_sets = prim.GetVariantSets()
+
+    # Convert do dict if we have a configclass object.
+    if variants is not isinstance(variants, dict):
+        variants = variants.__dict__
```

Comment:
Using this method might be better since it checks no dunder functions (like __del__) are captured

```suggestion
        variants = variants.to_dict()
```

Reply:
Actually, `@configclass` takes care of that as well even with `__dict__`.
![image](https://github.com/NVIDIA-Omniverse/orbit/assets/5803027/37295c52-a8f8-490d-8910-11856ccd053c)
But I can of course change it if you have strong feelings about this.